activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/connector/DefaultServiceTaskBehavior.java

```diff
@@ -21,26 +21,77 @@
 import org.activiti.bpmn.model.ServiceTask;
 import org.activiti.engine.delegate.DelegateExecution;
 import org.activiti.engine.impl.bpmn.behavior.AbstractBpmnActivityBehavior;
+import org.activiti.model.connector.ActionDefinition;
+import org.activiti.model.connector.ConnectorDefinition;
+import org.activiti.model.connector.VariableDefinition;
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.context.ApplicationContext;
 
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 public class DefaultServiceTaskBehavior extends AbstractBpmnActivityBehavior {
 
     private final ApplicationContext applicationContext;
     private final IntegrationContextBuilder integrationContextBuilder;
+    private final List<ConnectorDefinition> connectorDefinitions;
 
     public DefaultServiceTaskBehavior(ApplicationContext applicationContext,
-                                      IntegrationContextBuilder integrationContextBuilder) {
+                                      IntegrationContextBuilder integrationContextBuilder, List<ConnectorDefinition> connectorDefinitions) {
         this.applicationContext = applicationContext;
         this.integrationContextBuilder = integrationContextBuilder;
+        this.connectorDefinitions = connectorDefinitions;
     }
 
     @Override
     public void execute(DelegateExecution execution) {
-        Connector connector = applicationContext.getBean(getServiceTaskImplementation(execution),
-                                                         Connector.class);
-        IntegrationContext context = integrationContextBuilder.from(execution);
-        connector.execute(context);
-        execution.setVariables(context.getOutBoundVariables());
+        Connector connector;
+        IntegrationContext context;
+
+        String implementation = ((ServiceTask) execution.getCurrentFlowElement()).getImplementation();
+        String connectorId = StringUtils.substringBefore(implementation, ".");
```

Comment:
@Salaboy @ffazzini 
How about setting each as extensionElements of BPMN?
It is easier to understand than dot breaks.

Reply:
@daisuke-yoshimoto we are trying to move away from bpmn extension elements as much as we can. For this specific thing we are relying on a convention, but by default, the connectors mechanism will look for a registered bean as it is demonstrated in our activiti-examples repository.

Reply:
@Salaboy I see.